internal/app/streamer.go

```diff
@@ -94,3 +95,16 @@ func (b *BasinStreamer) sign(tx basincapnp.Tx) ([]byte, error) {
 
 	return signature, nil
 }
+
+func (b *BasinStreamer) push(ctx context.Context, table string, tx basincapnp.Tx, signature []byte) error {
+	if err := b.provider.Push(ctx, b.namespace, table, tx, signature); exc.IsType(err, exc.Disconnected) {
+		slog.Info("reconnecting")
+		if err := b.provider.Reconnect(); err != nil {
+			return fmt.Errorf("reconnect: %s", err)
+		}
+		return b.push(ctx, table, tx, signature)
```

Comment:
This is where we should add the exponential backoff? and maybe a timeout too? maybe worth adding a todo comment

Reply:
nice, added a todo. right now, it will only try reconnecting one time